pkg/apis/metrics/types.go

```diff
@@ -16,14 +16,427 @@ limitations under the License.
 
 package metrics
 
-import "k8s.io/kubernetes/pkg/api/unversioned"
+import (
+	"time"
 
-// Placeholder top-level node resource metrics.
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// RawNode holds node-level unprocessed sample metrics.
 type RawNode struct {
 	unversioned.TypeMeta `json:",inline"`
+	// Standard list metadata, applying to the lists of stats.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	unversioned.ListMeta `json:",inline"`
+	// Reference to the measured Node.
+	NodeRef api.ObjectReference `json:"nodeRef,omitempty"`
+	// Overall machine metrics.
+	Machine RawContainer `json:"machine,omitempty"`
+	// Metrics of system components.
+	SystemContainers []RawContainer `json:"systemContainers,omitempty"`
 }
 
-// Placeholder top-level pod resource metrics.
+// RawPod holds pod-level unprocessed sample metrics.
 type RawPod struct {
 	unversioned.TypeMeta `json:",inline"`
+	// Standard list metadata, applying to the lists of stats.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	unversioned.ListMeta `json:",inline"`
+	// Reference to the measured Pod.
+	PodRef api.ObjectReference `json:"podRef,omitempty"`
+	// Metrics of containers in the measured pod.
+	Containers []RawContainer `json:"containers,omitempty"`
+}
+
+// RawContainer holds container-level unprocessed sample metrics.
+type RawContainer struct {
+	// Reference to the measured container.
+	Name string `json:"name,omitempty"`
+	// Describes the container resources.
+	Spec ContainerSpec `json:"spec,omitempty"`
+	// Historical metric samples gathered from the container.
+	Stats []ContainerStats `json:"stats,omitempty"`
+}
+
+type ContainerSpec struct {
+	// Time at which the container was created.
+	CreationTime time.Time `json:"creationTime,omitempty"`
+
+	// Other names by which the container is known within a certain namespace.
+	// This is unique within that namespace.
+	Aliases []string `json:"aliases,omitempty"`
+
+	// Namespace under which the aliases of a container are unique (not a Kubernetes namespace).
+	// An example of a namespace is "docker" for Docker containers.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Metadata labels associated with this container.
+	Labels map[string]string `json:"labels,omitempty"`
```

Comment:
As with namespace, these aren't kubernetes labels (they're docker labels). Added a comment to that effect.